### PR TITLE
Remove unnecessary render scissor from ButtonWidget

### PR DESCRIPTION
**Pull request by dragunoff**
See https://github.com/OpenRA/OpenRA/pull/16552#discussion_r298548981

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/ButtonWidget.cs [hunk 1]
```diff
@@ -251,13 +251,9 @@ public override void Draw()
 			var hover = Ui.MouseOverWidget == this || Children.Any(c => c == Ui.MouseOverWidget);
 			DrawBackground(rb, disabled, Depressed, hover, highlighted);
 
-			// Scissor when the text overflows
-			var isTextOverflowing = font.Measure(text).X > contentWidth;
-			if (isTextOverflowing && TooltipContainer != null)
+			// Truncate when the text overflows
+			if (font.Measure(text).X > contentWidth && TooltipContainer != null)
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/ButtonWidget.cs]: This check is still dodgy in general: `text` may change between frames and if it hits this limit even once it will bind the tooltip for all future time.

Comment (pchote) [reply OpenRA.Mods.Common/Widgets/ButtonWidget.cs]: It would probably be better to revert the `ButtonWidget` changes completely and implement the truncation/tooltip logic in the `SettingsLogic` instead. You can guarantee the way your labels behave there, which saves you from having to guard against these more difficult dynamic cases.

**Comment by dragunoff** [comment]
Updated with @pchote's suggestion. 

But there is something subtle going on here (or I'm missing something obvious) because sometimes the tooltip would stop appearing. I can't reliably reproduce that - it seems to be happening at random after some rebindings.

File: OpenRA.Mods.Common/Widgets/ButtonWidget.cs [hunk 2]
```diff
@@ -207,11 +207,11 @@ public override bool HandleMouseInput(MouseInput mi)
 
 		public override void MouseEntered()
 		{
-			if (TooltipContainer == null || GetTooltipText() == null)
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/ButtonWidget.cs]: What is the motivation for these changes?

**Comment by pchote** [comment]
Problems with tooltips not showing when the UI changes underneath the mouse cursor are usually avoided by calling `Ui.ResetTooltips()` after changing the UI.

File: OpenRA.Mods.Common/Widgets/ButtonWidget.cs (hunk 2 above)
Comment (dragunoff) [reply OpenRA.Mods.Common/Widgets/ButtonWidget.cs]: It was crashing with a `System.NullReferenceException`. So I copied the logic from `LabelWithTooltipWidget`. The diffrence is in the null-check (`GetTooltipText()` vs `GetTooltipText`).

**Comment by dragunoff** [comment]
This is now based on #16759

I tried adding `Ui.ResetTooltips()` to several places in `SettingsLogic.cs` but to no avail. The tooltip would stop appearing after a few remaps.

**Comment by teinarss** [comment]
Rebase needed!

**Comment by dragunoff** [comment]
Rebased.

**Review by pchote — APPROVED** [review]
LGTM. I filed #16812 to cover the other issues, which can be done in a separate PR.

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/f510f8e78df8dc3c938f6f8590a151f56328a014...e6c878e0bf50904ae010b9d4abc8c19adcd4ad97)
